src/main/surfrdUtilsMod.F90

```diff
@@ -58,12 +58,15 @@ subroutine check_sums_equal_1(arr, lb, name, caller, ier, sumto)
 
     ub = ubound(arr, 1)
     allocate(TotalSum(lb:ub))
-    TotalSum = 1._r8
-    if ( present(sumto) ) TotalSum = sumto
     if( present(ier) ) ier = 0
     found = .false.
 
-    do nl = lbound(arr, 1), ub
+    do nl = lb, ub
+       if ( present(sumto) ) then
+          TotalSum(nl) = sumto(nl)
+       else
+          TotalSum(nl) = 1._r8
```

Comment:
As previously written, the code sometimes populated TotalSum with NaNs.

My changes get past this issue, but a new one has surfaced, so the pursuit continues.

Reply:
From today's stand-up conversation:
@billsacks suggested returning lines 60-69 to how they were.